### PR TITLE
adding Solidity Dev Survey Results 2020

### PR DESCRIPTION
**Pull request by franzihei**
~~- will probably still add a few more points to the summary on Monday~~
~~- will review with a fresh pair of eyes on Monday :)~~

Edit: Incorporated feedback received so far, adjusted a few graphs, adjusted summary part, finalized proof-reading from my side. Ready to be reviewed!

WRT the graphs: I know some of the legends are still hard to read (I edited some of them for better readability), but you can always right click on the image to view it larger in a separate tab.

### CODE REVIEW

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+The process of how to contribute to Solidity is clear or somewhat clear to 41% of respondents while 32% do not agree and hence say that the contribution process is unclear to them. 
+
+⚠️ _Note that there was a comparatively high share of "I don't know" replies to the last three statements._
+
+![Community Statements](/img/2021/01/communitystatements.png)
+
+#### Language Design
+
+Almost 85% of respondents state that they have never participated in any Solidity language design related activity. 2/3 of them say they didn't do so because they don't know how, while 1/3 is simply not interested in it.
```
Comment (hrkrshnn) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:335]:
```suggestion
Almost 85% of respondents state that they have never participated in any Solidity language design related activity. Around two third of them say they didn't do so because they don't know how, while one third is simply not interested in it.
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+![Language Design](/img/2021/01/langdesignparticipation.png)
+
+#### Staying Up-To-Date
+
+57% prefer Twitter to stay up-to-date about things. 21% like mailing lists, 14% prefer Reddit and 6% are in favor of forums.
+
+![Staying Up-To-Date](/img/2021/01/uptodate.png)
+
+To stay in the loop with Solidity updates like new releases, security alerts and announcements, most respondents said they follow Solidity on Twitter, follow the Solidity GitHub release page and/or the Solidity blog. Also mentioned but less popular are the r/ethdev Reddit, the Solidity documentation and the Week in Ethereum newsletter.
```
Comment (hrkrshnn) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:347]: Should we point to links here?

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+Some of the participants claiming to not do any of the above explained they mostly get informed by colleagues or friends or simply don't stay up-to-date at all.
```
Comment (hrkrshnn) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:349]:
```suggestion
Some of the participants claiming to not do any of the above explained that they mostly get informed by colleagues or friends or simply don't stay up-to-date at all.
```

**Review by hrkrshnn — COMMENTED** [review]
Many of the suggestions are subjective, so not necessary to incorporate.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+We were overwhelmed by the high quality of the submissions and are happy to extract important insights from your input.
+
+In this post, we'll be summarizing and commenting on the results of the survey. 
+
+Please note that none of the questions in the survey were mandatory. Hence, the amount of replies may vary per question.
+
+### Summary & Notable Insights
+
+- **Survey Audience**: 193 developers from all over the world replied to this survey, calling 48 different countries their current home. With roughly 20%, the big majority of respondents live in the United States of America, followed by India (8%) and Germany (7%). The majority of respondents speaks English at work.
```
[the remark this answers is not in the document]
Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:20]: I don't see why there should be a comma, but would use `speak` instead of `speak`, although I'm not sure about it.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+- **Solidity Experience**: 43% of respondents rate themselves Solidity experts with an expertise level of 8 (out of 10) or higher.
+- **Solidity Developer Experience**: While noting many things that could still improve, there was an overall positive/good sentiment towards Solidity. Most respondents believed that the Solidity developer experience had some or much improvement over the last year.
+- **Solidity's Challenges**: The biggest challenge or problem in developer experience for most respondents is the availability and quality of tooling. Also seen as critical aspects in order to foster adoption are error handling, IDE integrations and better documentation.
+- **Most Liked, Dreaded and Anticipated**: Topics with room for imporovement that came up a lot: Error handling, debugging, the need for more tools and libraries, breaking dependencies.
```
[the remark this answers is not in the document]
Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:25]: This might read as `need for breaking dependenciens` :), but I think it is better with `and`.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+- **Contributing and Language Design**: There was surprisingly little interest and engagement in language design topics, partly because developers didn't know how to get involved, partly because they were not interested. (We will work on this! Efforts to foster collaborative language design are under way, we'll share more soon.)
+- **Community**: Almost 17% of participants in this survey do not interact with other Solidity developers at all!
+
+### (1) Survey Audience and Developer Profiles
+
+Firstly, let's have a look at the developers that participated in the survey. In total, we received **193 responses from 48 different countries**.
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:31]: `-ly`?

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+- **Solidity's Challenges**: The biggest challenge or problem in developer experience for most respondents is the availability and quality of tooling. Also seen as critical aspects in order to foster adoption are error handling, IDE integrations and better documentation.
+- **Most Liked, Dreaded and Anticipated**: Topics with room for imporovement that came up a lot: Error handling, debugging, the need for more tools and libraries, breaking dependencies.
+- **Contributing and Language Design**: There was surprisingly little interest and engagement in language design topics, partly because developers didn't know how to get involved, partly because they were not interested. (We will work on this! Efforts to foster collaborative language design are under way, we'll share more soon.)
+- **Community**: Almost 17% of participants in this survey do not interact with other Solidity developers at all!
+
+### (1) Survey Audience and Developer Profiles
+
+Firstly, let's have a look at the developers that participated in the survey. In total, we received **193 responses from 48 different countries**. 
+
+![Survey Participants World Map](/img/2021/01/world.png)
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:33]: Would it be much effort to use a different projection than mercator? :)

Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:33]: Also, the file name is `World.png`, while the link here uses `world.png`.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Residency
+
+⚠️ _Note that the fact that this survey has only been shared in English is an important factor to consider when interpreting results regarding language and country of residency._
+
+The respondents stated to live in Algeria, Argentina, Australia, Austria, Belgium, Brazil, Bulgaria, Canada, China, Costa Rica, Denmark, Finland, France, Germany, Ghana, Greece, Honduras, Hong Kong, India, Indonesia, Iran, Ireland, Israel, Italy, Japan, Lithuania, Malaysia, Mexico, Netherlands, New Zealand, Nigeria, Pakistan, Philippines, Poland, Portugal, Romania, Russia, Serbia, Singapore, South Africa, Spain, Switzerland, Taiwan, Thailand, Timor-Leste, Ukraine, United Kingdom and United States of America.
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:39]: Could not resist but think of https://www.youtube.com/watch?v=AcS3NOQnsQM here.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Residency
+
+⚠️ _Note that the fact that this survey has only been shared in English is an important factor to consider when interpreting results regarding language and country of residency._
+
+The respondents stated to live in Algeria, Argentina, Australia, Austria, Belgium, Brazil, Bulgaria, Canada, China, Costa Rica, Denmark, Finland, France, Germany, Ghana, Greece, Honduras, Hong Kong, India, Indonesia, Iran, Ireland, Israel, Italy, Japan, Lithuania, Malaysia, Mexico, Netherlands, New Zealand, Nigeria, Pakistan, Philippines, Poland, Portugal, Romania, Russia, Serbia, Singapore, South Africa, Spain, Switzerland, Taiwan, Thailand, Timor-Leste, Ukraine, United Kingdom and United States of America. 
+
+With roughly 20%, the big majority of respondents live in the United States of America, followed by India (8%) and Germany (7%).
+
+![Survey Participants List of Countries](/img/2021/01/residency.png)
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:43]: Same capitalization problem in the file name, I guess it's everywhere.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Language
+
+The developers speak a wide variety of different languages as their primary language, with English being by far the most popular one. German, Spanish, Russian and French follow with similar levels.
+
+![Primary Language](/img/2021/01/langprimary.png)
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:49]: Why is this one sorted by language instead of "bar length"?

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 11]
```diff
@@ -0,0 +1,359 @@
...
+
+With 86%, the vast majority of respondents speak English at work. 
+
+Almost 90% of respondents are okay with reading the Solidity documentation in English while roughly 10% would prefer it to be translated into their native language. Some of the respondents who chose "native language" stated English as their primary spoken language though, which results in 95% of participants preferring English for the documentation.
+
+Bear in mind that this survey has only been conducted in English, which heavily influences the outcome of this question. This result will hence not have too much influence on the revamped translation efforts for the Solidity documentation and we will further investigate this topic.
```
[the remark this answers is not in the document]
Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:55]: I think it's still valid to re-state this here as is.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 12]
```diff
@@ -0,0 +1,359 @@
...
+
+The majority of respondents (68%) said they work in the technology industry. 14% work in the financial services industry. Less relevant, but also selected several times, were the energy sector, health care, creative, media & gaming and transportation.
+
+![Industry Overview](/img/2021/01/industry.png)
+
+#### Coding Experience
+
+The overall, non-Solidity-related, professional coding experience of respondents was impressive. 48% of respondents said they coded as part of their jobs for 6 years or more. Only roughly 7% stated that they have less than 1 year of coding experience. 
+
+![Professional Coding Experience](/img/2021/01/codingexperience.png)
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:78]: Again, if it's not too much of a hassle, everything except the "no coding experince" result could be stacked on the same bar because each answer "includes" all of the previous ones.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 13]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Coding Preferences
+
+We've asked the participants what is most important to them when choosing a programming language. The question had a free text reply field, hence we clustered the responses into overarching categories. 
+
+The most important aspect, stated by many respondents, was the fact that the programming language has to fit the task, it has to "get the job done".
+
+Also ranked very highly were tooling, documentation and community, followed by security of the language, libraries, availability of good resources and learning material as well as a clear or familiar syntax. 
+
+Less frequently mentioned, but still relevant, were descriptive error messages / good debugging, readability, performance and strong types.
```
[the remark this answers is not in the document]
Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:108]: Less frequently does not necessarily mean infrequently...

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 14]
```diff
@@ -0,0 +1,359 @@
...
+
+The majority of respondents use Solidity on a weekly (31%) or even daily basis (41%). Almost 18% use it monthly, while the remaining 9% stated to rarely code in Solidity.
+
+![Frequency of Usage](/img/2021/01/howoftensol.png)
+
+Most respondents are currently using 0.6.x or 0.7.x versions of Solidity. A smaller share still uses 0.5.x versions or has been trying out the recently released 0.8.0 already.
+
+A few people are not aware of the version they are using or are still using very old versions from the 0.4.x series. 
+
+🚨 Please make sure to frequently update your code since several important bug fixes and security improvements have been added since 0.4.x!
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:189]: Could add a link to here: https://github.com/ethereum/solidity/blob/develop/docs/bugs_by_version.json#L922

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 15]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Overall Solidity Developer Experience
+
+62% believe that the Solidity developer experience had some or even a lot of improvement in the last year. 15% saw a slight improvement while 9% thought that there was no change or that the Solidity developer experience got worse.
+
+![Overall Developer Experience](/img/2021/01/devexperience.png)
+
+The biggest challenge or problem in developer experience for most respondents is the availability and quality of tooling. 
+
+Also seen as critical aspects in order to foster adoption are error handling, IDE integrations and better documentation.
```
[the remark this answers is not in the document]
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:212]: I don't think these commas are necessary.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 16]
```diff
@@ -0,0 +1,359 @@
...
+The currently most dreaded aspect about Solidity is clearly debugging, followed by a lack of Solidity libraries. 
+
+Respondents also complain about inheritance conflicts and the need to write more specific code in assembly / low-level code. 
+
+Other aspects and features that have been mentioned repeatedly as dreaded or disliked are:
+- Error handling
+- Memory management
+- Frequency of breaking changes and dependencies breaking
+- Contract & stack size limits
+- Stack too deep
```
Comment (chriseth) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:275]: stack size limit and stack too deep is the same thing, I would say

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 9 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:43]: yes will adjust everywhere

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 7 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md]: yes, it would be quite some work to use a different projection since Mercator is the only one available for Google Sheet's Geo Charts

Comment (hrkrshnn) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md]: Why don't you like Mercantor? @chriseth

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:20]: I think the majority is the subject, so even though it sounds weird it should be `speaks`

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 7 above)
Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md]: Due to its distortions, countries close to the equator are drawn smaller than they are and thus it "favours" countries on the northern hemisphere, i.e. the "western" countries. Also, greenland and northern canada just look really weird. Maybe we can crop a bit more towards the north pole, @franzihei ?

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:20]: Learnt something today! It's "the majority speaks" but "a majority speak" - https://www.enago.com/academy/understanding-the-usage-of-majority/

[the remark this answers is not in the document]
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:20]: Now I am confused. 😅

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 17]
```diff
@@ -0,0 +1,358 @@
...
+
+The majority of respondents use Solidity on a weekly (31%) or even daily basis (41%). Almost 18% use it monthly, while the remaining 9% stated that they rarely code in Solidity.
+
+![Frequency of Usage](/img/2021/01/HowOftenSol.png)
+
+Most respondents are currently using 0.6.x or 0.7.x versions of Solidity. A smaller share still uses 0.5.x versions or has been trying out the recently released 0.8.0 already.
+
+A few people are not aware of the version they are using or are still using very old versions from the 0.4.x series. 
+
+🚨 Please make sure to frequently update your code since [several important bug fixes](https://github.com/ethereum/solidity/blob/develop/docs/bugs_by_version.json#L922) and security improvements have been added since 0.4.x!
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:189]: It's pretty interesting that the list of important bugs becomes nearly empty with 0.7.x. It doesn't seem to have anything to do with release frequency (0.5.x had tons of them even with more than one release a month sometimes). Maybe it's the effect of better code review coverage with larger team?

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 18]
```diff
@@ -0,0 +1,358 @@
...
+
+![IDEs / Tooling](/img/2021/01/IDE.png)
+
+When asked if they are using an Ethereum-specific development environment, only 3% say that they are not using one. Truffle is the most prominent with 35%, closely followed by Remix (29%) and Hardhat (23%).
+
+![Ethereum-specific IDEs / Tooling](/img/2021/01/EthIDE.png)
+
+#### Overall Solidity Developer Experience
+
+Around 62% believe that the Solidity developer experience had some or even a lot of improvement in the last year. 15% saw a slight improvement while 9% thought that there was no change or that the Solidity developer experience got worse.
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:206]:
```suggestion
Around 62% believe that the Solidity developer experience had some or even a lot of improvement in the last year. 15% saw a slight improvement while only 9% thought that there was no change or that the Solidity developer experience got worse.
```
Maybe it's worth emphasizing that it's a pretty small number.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 19]
```diff
@@ -0,0 +1,358 @@
...
+
+![Overall Developer Experience](/img/2021/01/DevExperience.png)
+
+The biggest challenge in developer experience for most respondents is the availability and quality of tooling. 
+
+Also seen as critical aspects in order to foster adoption are error handling, IDE integrations and better documentation.
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:212]: This sentence seems a bit hard to parse to me. How about this?
```suggestion
Other aspects seen as critical in fostering language adoption are: Error handling, IDE integrations and better documentation. 
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 20]
```diff
@@ -0,0 +1,358 @@
...
+
+Also seen as critical aspects in order to foster adoption are error handling, IDE integrations and better documentation. 
+
+Less often voted for but still relevant are the increasing complexity of the language, availability of generics, improvements in compilation time and debugging as well as improvements in gas cost, storage handling and security.
+
+![Solidity's Challenges](/img/2021/01/Challenges.png)
+
+#### Importance Ranking of Future Features
+
+We've asked the participants of the survey to rank selected features under discussion according to their importantance.
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:220]:
```suggestion
We've asked the participants of the survey to rank selected features under discussion according to their importance.
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 21]
```diff
@@ -0,0 +1,358 @@
...
+
+Gas optimizations were ranked as most important, followed by interfaces to external debugging tools. The SMTChecker, generics and fixed point types were ranked similarly important. 
+
+Explicitly copying semantics, enums with data, functions at file level and WebAssembly output were evaluated as less critical by the participants.
+
+![Future Features Ranking](/img/2021/01/FutureFeatures.png)
+
+### (4) Most Liked, Dreaded and Anticipated Features
+
+We've asked survey participants what they like and dislike most about Solidity and what they would like to see the most in upcoming Solidity versions. Those questions had free-text reply fields so we've summarized the replies and are listing the most mentioned features and aspects below.
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:230]:
```suggestion
We've asked survey participants what they like and dislike most about Solidity and what they would like to see the most in upcoming Solidity versions. Those questions had free-form text reply fields so we've summarized the replies and are listing the most mentioned features and aspects below.
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 22]
```diff
@@ -0,0 +1,358 @@
...
+
+Furthermore, the developers appreciated the improvements in available tooling, especially highlighting Remix and Hardhat several times. 
+
+Another aspect the respondents appreciate is that Solidity is "easy to learn" and "simple to code".
+
+Several people stated that Solidity is the right tool for what they are working on and that they like the focus on smart contracts and Solidity being a "special purpose tool" for it.
+
+Other features and aspects that have been mentioned repeatedly as cherished and appreciated are:
+- Mappings
+- Human readability
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:244]:
```suggestion
- Human-readability
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 23]
```diff
@@ -0,0 +1,358 @@
...
+- Explicitness
+- Inheritance
+- Compact code size
+- Syntax
+- Steady improvement of the language
+
+
+#### Most Dreaded or Disliked Features
+
+The currently most dreaded aspect about Solidity is clearly debugging, followed by a lack of Solidity libraries.
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:266]:
```suggestion
The currently most dreaded aspect of Solidity is clearly debugging, followed by a lack of Solidity libraries. 
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 24]
```diff
@@ -0,0 +1,358 @@
...
+
+Respondents also complain about inheritance conflicts and the need to write more specific code in assembly / low-level code. 
+
+Other aspects and features that have been mentioned repeatedly as dreaded or disliked are:
+- Error handling
+- Memory management
+- Frequency of breaking changes and dependencies breaking
+- Contract & stack size limits
+- Inconsistency with other languages
+- ABI encode
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:276]:
```suggestion
- ABI encoder
```
Or `ABI encoding`?

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 25]
```diff
@@ -0,0 +1,358 @@
...
+Other anticipated and frequently mentioned features were:
+- Gas efficiency optimization features
+- Generics
+- console.log()
+- Dynamic arrays
+- Better tooling
+- Better IDEs
+- LSP support
+- Structs
+- Rich revert reasons/ error codes
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:302]:
```suggestion
- Rich revert reasons / error codes
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 26]
```diff
@@ -0,0 +1,358 @@
...
+- Standard library
+- Custom optimization features
+- Release v.1.0.0
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:305]:
```suggestion
- Release v1.0.0
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 27]
```diff
@@ -0,0 +1,358 @@
...
+- SMTChecker
+- Stable ABIEncoderV2
+- No more stack too deep
+- Decimals
+- Over- and underflow handling
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:310]: I.e. SafeMath :)

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 28]
```diff
@@ -0,0 +1,358 @@
...
+
+Other anticipated and frequently mentioned features were:
+- Gas efficiency optimization features
+- Generics
+- console.log()
+- Dynamic arrays
+- Better tooling
+- Better IDEs
+- LSP support
+- Structs
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:301]: WTF. We do have structs :)

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 29]
```diff
@@ -0,0 +1,358 @@
...
+The most anticipated feature by far was "no more SafeMath". SafeMath by default has been [introduced with 0.8.0](https://blog.soliditylang.org/2020/12/16/solidity-v0.8.0-release-announcement/) and hence is not an issue anymore. 🥳
+
+The second most anticipated feature was better debugging and more debugging tools. 
+
+Also ranked high was better documentation, especially with a focus on more tutorials and code examples.
+
+Other anticipated and frequently mentioned features were:
+- Gas efficiency optimization features
+- Generics
+- console.log()
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:296]: Some of your lists have items ending with dots, some don't. I think it should be consistent.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 30]
```diff
@@ -0,0 +1,358 @@
...
+- Rich revert reasons/ error codes
+- Standard library
+- Custom optimization features
+- Release v.1.0.0
+- SMTChecker
+- Stable ABIEncoderV2
+- No more stack too deep
+- Decimals
+- Over- and underflow handling
+- Fixed point types
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:311]: I'd group decimals together with fixed-point types. Maybe people even meant the same thing by it. Technically fixed-point numbers stored as decimal numbers (rather than binary ones) would be a separate feature but I suspect that people were just casually referring to fractional numbers as decimals.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 31]
```diff
@@ -0,0 +1,358 @@
...
+- WASM
+
+### (5) Community
+
+More than 55% of respondents interact with other Solidity developers on a regular basis, while 27% only claim to do so rarely. Almost 17% do not interact with other Solidity developers at all.
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:316]:
```suggestion
More than 55% of respondents interact with other Solidity developers on a regular basis, while 27% claim to do so only rarely. Almost 17% do not interact with other Solidity developers at all.
```

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 32]
```diff
@@ -0,0 +1,358 @@
...
+
+The process of how to contribute to Solidity is clear or somewhat clear to 41% of respondents while 32% do not agree and hence say that the contribution process is unclear to them. 
+
+⚠️ _Note that there was a comparatively high share of "I don't know" replies to the last three statements._
+
+![Community Statements](/img/2021/01/CommunityStatements.png)
+
+#### Language Design
+
+Almost 85% of respondents state that they have never participated in any Solidity language design related activity. Around two third of them say they didn't do so because they don't know how, while one third is simply not interested in it.
```
Comment (cameel) [inline _posts/2021-01-25-solidity-developer-survey-2020-results.md:334]:
```suggestion
Almost 85% of respondents state that they have never participated in any Solidity language design related activity. Around two thirds of them say they didn't do so because they don't know how, while one third is simply not interested in it.
```

**Review by cameel — COMMENTED** [review]
Great work on the article! I found some typos and have some minor suggestions but overall it looks pretty good.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 33]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Operating System
+
+Interestingly, Linux and macOS seem equally popular, both getting roughly 40% of the repondents' votes. Circa 18% use Windows, while a part of those users also state to use another OS in addition to Windows. There are also several developers using both Linux and macOS, or even all three operating systems.
+
+![Operating System](/img/2021/01/operatingsystem.png)
+
+#### Solidity Usage
+
+Almost all of the respondents use Solidity for personal projects outside of work (93%), while 68% say they also use Solidity at work and code in another programming language at work.
```
[the remark this answers is not in the document]
Comment (cameel) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:88]: Maybe it would be clearer if it said that you could choose more than one answer?

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 34]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Language
+
+The developers speak a wide variety of different languages as their primary language, with English being by far the most popular one. German, Spanish, Russian and French follow with similar levels.
+
+![Primary Language](/img/2021/01/langprimary.png)
+
+With 86%, the vast majority of respondents speak English at work. 
+
+Almost 90% of respondents are okay with reading the Solidity documentation in English while roughly 10% would prefer it to be translated into their native language. Some of the respondents who chose "native language" stated English as their primary spoken language though, which results in 95% of participants preferring English for the documentation.
```
[the remark this answers is not in the document]
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:53]: 89 speak English as their primary language, 11 of them stated that they would prefer the documentation in their native language and didn't just choose "English", even though their native language seems to be English, so I've added 11 + 172 (people who chose English for documentation) and divided by 193 which is 94,82%.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 6 above)
Comment (cameel) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md]: TIL [Some dictionaries don't even recognize `firstly` as a word](https://www.grammarly.com/blog/firstly/). I'm using it all the time :)

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 33 above)
[the remark this answers is not in the document]
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:88]: tried to simplify it a bit

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 34 above)
[the remark this answers is not in the document]
Comment (hrkrshnn) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:53]: Ah. Makes sense. I somehow assumed that you could get the 95% from the other two numbers.

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md [hunk 35]
```diff
@@ -0,0 +1,359 @@
...
+
+The Solidity documentation is the most prominent learning material and primary resource for most respondents to learn Solidity. 
+
+The practices of reading the open-source Solidity code of big projects as well as watching coding tutorial videos were frequently mentioned.
+
+Another prominent way to get started learning Solidity is the CryptoZombies coding game. 
+
+Many respondents used the Ethereum StackExchange to find answers to their questions. 
+
+Massive open online courses (MOOCs) like Udacity, Udemy and university courses were mentioned by several respondents followed by active coding approaches like "learning by coding" using Remix and the OpenZeppelin contracts as well as by going through the ConsenSys Academy.
```
[the remark this answers is not in the document]
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:142]: added a couple of links!

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 10 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md]: changed the language graphs to be sorted by length

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 20 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:220]: lol 🤣

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 27 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:310]: right, I'll remove it as it is covered on the top of the paragraph :)

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 28 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:301]: yup that doesn't make sense here maybe a copy paste error

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 29 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md:296]: right, will add the dots everywhere for better accessibility. :)

File: _posts/2021-01-25-solidity-developer-survey-2020-results.md (hunk 12 above)
Comment (franzihei) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md]: I'm not sure what you're suggesting exactly? That there are only two bars? I think that would be quite hard to read

Comment (chriseth) [reply _posts/2021-01-25-solidity-developer-survey-2020-results.md]: Ok, was just a suggestion!
